logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -15,6 +15,7 @@ def register
   end
 
   def run(queue)
+    @logger.debug("check log4j fish tagging: [" + ThreadContext.get("plugin.id") + "]")
```

Comment:
are these debug lines in the spec still necessary or useful?

Reply:
yes it is need to prove that the ThreadContext contains the fish tag for the plugin, so inside the plugin I log the tag and then in the `expect` verify that the log contains the tag.

Reply:
Probably you are right, I can remove the "compiler way" that insert java code so set/unset the value in the ThreadContext, and move into the `doOutput` of `AbstractOutputDelegatorExt` (from `OutputDelegatorExt`), having a uniform way to set the plugin name for Ruby and Java plugins. What do you think?